File: Java/Prototype/src/main/java/com/hydron/sample/pojo/Foo.java
package com.hydron.sample.pojo;

import lombok.*;

import java.time.Instant;

@NoArgsConstructor
@AllArgsConstructor
@Builder
@Getter
@ToString
public class Foo implements ExampleData {
    private String name;
    private String identifier;
    private Instant timestamp;
    private static final String text = "%s [%s] - Foo says \"Hello, %s!\"";

    @Override
    public String getMessage() {
        return String.format(text, this.identifier, this.timestamp, this.name);
    }
}

File: Rust/modular_library/src/foo.rs
use serde::{Deserialize, Serialize};

use crate::example_data::ExampleData;

#[derive(Serialize, Deserialize)]
pub struct Foo {
    name: String,
    identifier: String,
    timestamp: String,
}

impl ExampleData for Foo {
    fn new(name: &str, identifier: &str, timestamp: String) -> Self {
        return Self {
            name: String::from(name),
            identifier: String::from(identifier),
            timestamp,
        };
    }

    fn get_message(&self) -> String {
        return format!("{} [{}] - Foo says \"Hello, {}!\"", self.identifier, self.timestamp, self.name);
    }
}
